controller/api/destination/watcher/endpoints_watcher.go

```diff
@@ -886,52 +886,57 @@ func (pp *portPublisher) endpointSliceToAddresses(es *discovery.EndpointSlice) A
 		}
 
 		if endpoint.TargetRef.Kind == endpointTargetRefPod {
-			for _, IPAddr := range endpoint.Addresses {
-				address, id, err := pp.newPodRefAddress(resolvedPort, IPAddr, endpoint.TargetRef.Name, endpoint.TargetRef.Namespace)
-				if err != nil {
-					pp.log.Errorf("Unable to create new address:%v", err)
-					continue
-				}
-				err = SetToServerProtocol(pp.k8sAPI, &address, resolvedPort)
-				if err != nil {
-					pp.log.Errorf("failed to set address OpaqueProtocol: %s", err)
-					continue
-				}
+			if len(endpoint.Addresses) < 1 {
+				continue
+			}
+			IPAddr := endpoint.Addresses[0]
+			address, id, err := pp.newPodRefAddress(resolvedPort, IPAddr, endpoint.TargetRef.Name, endpoint.TargetRef.Namespace)
+			if err != nil {
+				pp.log.Errorf("Unable to create new address:%v", err)
+				continue
+			}
+			err = SetToServerProtocol(pp.k8sAPI, &address, resolvedPort)
+			if err != nil {
+				pp.log.Errorf("failed to set address OpaqueProtocol: %s", err)
+				continue
+			}
```

Comment:
Can we think through any situation in which this could  be correct incorrect for a Pod?

Reply:
To process only one address, you mean? Or? 

The upstream contract for an endpointslice does mention that clients may choose to process as many as they want, but they're assumed to be fungible so processing the first one is enough.

https://github.com/kubernetes/api/blob/f371fc655f9e261a8b523f396fc8ccce9bfa27fe/discovery/v1/types.go#L78-L85